### PR TITLE
PMM-10814: List active service types

### PR DESCRIPTION
**Pull request by michal-kralik**
PMM-10814

Build: SUBMODULES-2813

### CODE REVIEW

**Review by BupycHuk — APPROVED** [review]

**Comment by BupycHuk** [comment]
@michal-kralik Can we write a few API tests?

File: managed/models/service_helpers.go [hunk 1]
```diff
@@ -123,6 +124,33 @@ func FindServices(q *reform.Querier, filters ServiceFilters) ([]*Service, error)
...
+	rows, err := q.Query(query)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	defer func() {
+		if err := rows.Close(); err != nil {
+			logrus.Debug(err)
+		}
+	}()
```
Comment (artemgavrilov) [inline managed/models/service_helpers.go:139]: Here we will overwrite error returned on line 145

File: managed/models/service_helpers.go [hunk 2]
```diff
@@ -123,6 +124,33 @@ func FindServices(q *reform.Querier, filters ServiceFilters) ([]*Service, error)
...
+
+	res := []ServiceType{}
```
Comment (artemgavrilov) [inline managed/models/service_helpers.go:141]: https://dave.cheney.net/2014/05/24/on-declaring-variables
```suggestion
	var res []ServiceType
```

File: managed/models/service_helpers.go [hunk 3]
```diff
@@ -123,6 +124,33 @@ func FindServices(q *reform.Querier, filters ServiceFilters) ([]*Service, error)
...
+	for rows.Next() {
+		var serviceType ServiceType
+		if err := rows.Scan(&serviceType); err != nil {
```
Comment (artemgavrilov) [inline managed/models/service_helpers.go:144]:
```suggestion
		if err = rows.Scan(&serviceType); err != nil {
```

File: managed/models/service_helpers_test.go [hunk 4]
```diff
@@ -266,6 +266,15 @@ func TestServiceHelpers(t *testing.T) {
 		}})
 	})
 
+	t.Run("FindActiveServiceTypes", func(t *testing.T) {
+		q, teardown := setup(t)
+		defer teardown(t)
+
+		types, err := models.FindActiveServiceTypes(q)
+		assert.NoError(t, err)
+		assert.Equal(t, 6, len(types))
```
Comment (artemgavrilov) [inline managed/models/service_helpers_test.go]:
```suggestion
		assert.Len(t, types, 6)
```

File: managed/services/inventory/grpc/services_server.go [hunk 5]
```diff
@@ -88,6 +88,25 @@ func (s *servicesServer) ListServices(ctx context.Context, req *inventorypb.List
...
+	types, err := s.s.ListActiveServiceTypes(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	res := &inventorypb.ListActiveServiceTypesResponse{
+		ServiceTypes: make([]inventorypb.ServiceType, 0, len(types)),
+	}
+
+	res.ServiceTypes = append(res.ServiceTypes, types...)
```
Comment (artemgavrilov) [inline managed/services/inventory/grpc/services_server.go:105]: It seems we can just do it like this:
```suggestion
	res := &inventorypb.ListActiveServiceTypesResponse{
		ServiceTypes: types,
	}
```

File: managed/services/inventory/services.go [hunk 6]
```diff
@@ -76,6 +76,40 @@ func (ss *ServicesService) List(ctx context.Context, filters models.ServiceFilte
...
+func (ss *ServicesService) ListActiveServiceTypes(ctx context.Context) ([]inventorypb.ServiceType, error) {
+	var types []models.ServiceType
+	err := ss.db.InTransaction(func(tx *reform.TX) error {
+		var err error
+		types, err = models.FindActiveServiceTypes(tx.Querier)
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
```
Comment (artemgavrilov) [inline managed/services/inventory/services.go:91]: Let's keep it consistent with other parts
```suggestion
	e := ss.db.InTransaction(func(tx *reform.TX) error {
		var err error
		types, err = models.FindActiveServiceTypes(tx.Querier)
		return err
	})
	if e != nil {
		return nil, e
	}
```

File: managed/services/inventory/services.go [hunk 7]
```diff
@@ -76,6 +76,40 @@ func (ss *ServicesService) List(ctx context.Context, filters models.ServiceFilte
...
+			res = append(res, inventorypb.ServiceType_MONGODB_SERVICE) //nolint:nosnakecase
+		case models.PostgreSQLServiceType:
+			res = append(res, inventorypb.ServiceType_POSTGRESQL_SERVICE) //nolint:nosnakecase
+		case models.ProxySQLServiceType:
+			res = append(res, inventorypb.ServiceType_PROXYSQL_SERVICE) //nolint:nosnakecase
+		case models.HAProxyServiceType:
+			res = append(res, inventorypb.ServiceType_HAPROXY_SERVICE) //nolint:nosnakecase
+		case models.ExternalServiceType:
+			res = append(res, inventorypb.ServiceType_EXTERNAL_SERVICE) //nolint:nosnakecase
+		}
```
Comment (artemgavrilov) [inline managed/services/inventory/services.go:108]: It seems that we can just remove this switch-case and just add every `t` to result

File: managed/models/service_helpers.go (hunk 1 above)
Comment (artemgavrilov) [reply managed/models/service_helpers.go:139]: Ah, no, I'm wrong.

Comment (artemgavrilov) [reply managed/models/service_helpers.go:139]: BTW can we use different name for error variable here, to make it more obvious?

Comment (michal-kralik) [reply managed/models/service_helpers.go]: Sure

File: managed/services/inventory/grpc/services_server.go (hunk 5 above)
Comment (michal-kralik) [reply managed/services/inventory/grpc/services_server.go]: Great catch

File: managed/services/inventory/services.go (hunk 7 above)
Comment (michal-kralik) [reply managed/services/inventory/services.go:108]: `t` is of type `string` and `res` requires `int`. It appears to me there needs to be a way to map the values to each other.
How would you suggest we use just `t`? Maybe I'm missing something.

Comment (artemgavrilov) [reply managed/services/inventory/services.go:108]: Ah, yes, my bad. Nevermind.

**Review by artemgavrilov — APPROVED** [review]

**Pull request merged**